### ISSUE TITLE
 control panel 

### ISSUE THREAD

**Issue by gedw99** [HouzuoGuo/tiedot#94]
i noticed that in this release the web control panel is gone.

is the code still in the tree ? I would really like to use it and maybe extend it. useful to me.

**Comment by HouzuoGuo** [comment]
Very sorry about that, the control panel was served by this package called "go-rice" that does not work at all when packaged into an RPM. Also, when tiedot v3 came out, the API for handling document ID changed and several of us couldn't figure out how to fix it.

If you happen to have some extra time, you are certainly more than welcomed to create the next generation web CP for tiedot :+1:

**Comment by ktw** [comment]
Just curious, what exactly was the problem with the ID's? (Besides the RPM issue) 
I cloned the latest master, and added the control panel from the 32bit branch, and it seems to work just fine for creating data, and querying by ID. 
(I had some variables missing in main.go, and some imports that needed to be local, "github.com/HouzuoGuo/tiedot/..." -> "tiedot/...")

**Comment by HouzuoGuo** [comment]
about a year and half ago, tiedot used to store document ID along with other document attributes, so that the ID would be deserialised along with other document attributes when retrieved.

Starting from v3, the document ID is no present in the document content. on the web UI, if user attempts to create a document with an attribute called "_id" (or "id", I do not remember clearly), the attribute will be removed before HTTP API is called.

**Comment by ktw** [comment]
Yes I see. It is "id" which causes the problems, but "_id" is allowed though.

Do you mind if I try to fix it, or perhaps create a new control panel?
(I am not a designer, so it might end up being ugly)

**Comment by HouzuoGuo** [comment]
that's much appreciated! please feel free to bring it back in master, i added you as collaborator.

**Comment by ktw** [comment]
I am creating the control panel from scratch, so here is a quick status update:
- Re-enabled the user interface section
- Added functionality for localizing both html templates and errors (Using key / value pairs in JSON files)

I have created the following very basic pages: (No layout or styling yet)
- List collections
- List documents in a collection (Using paging, approximately 30 documents on each page)
- View / edit documents (No problems when saving "id" and "_id" values)

I am new to Go, and my spare time is rather limited, so it might take some time to complete.
But it's a great learning experience for me.

**Comment by HouzuoGuo** [comment]
Thank you soooo much for making the effort!

**Comment by corebreaker** [comment]
ktw, are you pushed your dev ?

**Comment by kerneltravel** [comment]
[bleve-explorer](https://github.com/blevesearch/bleve-explorer) is a panel for bleve, a json-support fulltext indexer.

**Comment by lukerodham** [comment]
Did this ever happen? I'm keen to help out if it did or start something if it didn't.

**Comment by HouzuoGuo** [comment]
@lukerodham not much progress has been made in regards to making the control panel application. Your help would be very much appreciated!
